File: 242-ValidAnagram/242-ValidAnagram.cpp
// Last updated: 6/10/2026, 7:25:26 PM
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> freq(26 , 0);
        for (auto c : s) freq[c - 'a']++;
        for (auto c : t) freq[c - 'a']--;
        for (auto i : freq) if (i != 0) return false;
        return true;
    }
};

File: 242-ValidAnagram/242-ValidAnagram.java
// Last updated: 6/10/2026, 7:25:30 PM
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] freq = new int[26];
        Arrays.fill(freq , 0);
        for (char c : s.toCharArray()) freq[c - 'a']++;
        for (char c : t.toCharArray()) {
            if (freq[c - 'a'] == 0) return false;
            freq[c - 'a']--;
        }
        for (int i=0 ; i<26 ; i++) {
            if (freq[i] > 0) return false;
        }
        return true;
    }
}